tests/unit_tests/epee_utils.cpp

```diff
@@ -1818,13 +1842,44 @@ TEST(parsing, unicode)
   std::string s;
   std::string::const_iterator si;
```

Comment:
This `si` variable seems unnecessary (just one more odd thing from this ancient code). You could replace the `si` parameter passing with `s.begin()`. The same goes for `i` in the other unit test.

Reply:
Ok maybe not... this function parameter is a non-const l-value reference, which can't be initialized from an r-value... sigh. One example of why code quality is important: always make non-modifiable non-sink function parameters const.